### PR TITLE
Added generate:toc script to genrate table of contents

### LINKED ISSUE #1528: The `INSTALLATION.md` file needs a newly generated table of contents
**Is your feature request related to a problem? Please describe.**
1. There are many missing table of contents (TOC) entries in the INSTALLATION.md file.
1. Many of the existing TOC entries are under an inappropriate parent heading

This increases the risk of installation instructions being overlooked.

**Describe the solution you'd like**
1. Add the missing table of contents (TOC) entries in the INSTALLATION.md file.
1. Adjust the resulting TOC entries under the appropriate parent headings
1. Ideally, the auto-generation of the TOC should be automatic as part of the `git` process

**Describe alternatives you've considered**
- N/A

**Approach to be followed (optional)**
- Consider using a markdown TOC autogenerator

**Additional context**
- N/A

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by Veer0x1**
**What kind of change does this PR introduce?**
This PR adds a script `generate:toc` to automatically generate table of contents for all the makrdown files in repository, for this command to work, there should we a comment of `<!-- toc -->` and `<!-- tocstop -->`. The table of content will be place under this comments. If this comments aren't present in any markdown file then there will be no change in those files. 
I have placed this comments in all the neccessary markdown files excepts some markdown files in .github/ directory coz there is no need of it. This done via a `markdow-toc` npm package. I have also added instruction on how to use this script in `CONTRIBUTING.md` file

**Issue Number:**

Fixes #1528 

**Did you add tests for your changes?**
NA

**Snapshots/Videos:**

**If relevant, did you update the documentation?**

**Summary**

**Does this PR introduce a breaking change?**
No

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**
Yes

### CODE REVIEW

**Review by palisadoes — CHANGES_REQUESTED** [review]
The files in the `.github` directory must not be updated with a TOC. It could affect the issue creation workflow

**Comment by Veer0x1** [comment]
> The files in the `.github` directory must not be updated with a TOC. It could affect the issue creation workflow

Only the readme in workflow directory is being update with TOC, so it shouldn't affect any other workflow, or Template

**Comment by palisadoes** [comment]
1. Only the markdown files in the root directory must be updated. Your change must not affect any other locations
1. The `.github` directory is out of scope for this issue.

**Comment by Veer0x1** [comment]
> 1. Only the markdown files in the root directory must be updated. Your change must not affect any other locations
> 2. The `.github` directory is out of scope for this issue.

Done, now the script doesn't changes the markdown files in `.github` directory

File: .github/ISSUE_TEMPLATE/bug-report.md [hunk 1]
```diff
@@ -4,9 +4,7 @@ about: Create a report to help us improve.
 title: Bug Report
 labels: bug
 assignees: ''
-
 ---
```
Comment (palisadoes) [inline .github/ISSUE_TEMPLATE/bug-report.md:7]: Do not include this file in the PR

File: .github/ISSUE_TEMPLATE/feature-request.md [hunk 2]
```diff
@@ -24,3 +22,10 @@ Add any other context or screenshots about the feature request here.
 
 **Potential internship candidates**
 Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359
```
Comment (palisadoes) [inline .github/ISSUE_TEMPLATE/feature-request.md:24]: Do not include this file in the PR

File: .husky/pre-commit [hunk 3]
```diff
@@ -18,5 +18,8 @@ npm run lint:fix
 
 npm run gen:schema
 
+# Generate table of contents for all markdown files
+npm run generate:toc
```
Comment (palisadoes) [inline .husky/pre-commit:22]: This should specify only the markdown in the root directory. How can that be done?

**Review by palisadoes — CHANGES_REQUESTED** [review]
See comments

File: .github/ISSUE_TEMPLATE/bug-report.md (hunk 1 above)
Comment (Veer0x1) [reply .github/ISSUE_TEMPLATE/bug-report.md:7]: I have ignored `.github` directory overall, it was from my last commit that added extra new line at end. I'll remove it.

**Review by palisadoes — CHANGES_REQUESTED** [review]
If that is the case, why have the .github directory files been re-added?

**Comment by Veer0x1** [comment]
> If that is the case, why have the .github directory files been re-added

> If that is the case, why have the .github directory files been re-added?

In my last commit the script has added trailing lines at the end of `md` files in `.github` directory, in new commit I just copy pasted the original content, thats why.

**Comment by palisadoes** [comment]
1. The files must not be included in the PR
1. You still haven't answered my question about how you will ensure that the TOC will only be applied to the markdown files in the root directory

**Comment by Veer0x1** [comment]
> 1. The files must not be included in the PR
> 2. You still haven't answered my question about how you will ensure that the TOC will only be applied to the markdown files in the root directory

To ensure that only markdown files in the root the script only searches in root as its mentioned in command that maxDept is 1
`"generate:toc": "find . -maxdepth 1 -name '*.md' -exec markdown-toc -i {} \\;"`

okay I'll completely remove the Template mardown from this PR.

**Comment by Veer0x1** [comment]
Let me raise a fresh PR.

**Pull request closed without merging**